harness/determined/_execution.py

```diff
@@ -65,6 +65,7 @@ def _make_local_execution_env(
     config: Optional[Dict[str, Any]],
     hparams: Optional[Dict[str, Any]] = None,
     limit_gpus: Optional[int] = None,
+    test_mode: bool = False,
```

Comment:
Can we make this a required parameter?  It's not really optional; we always need to know if the environment is in test mode, the same as we always need to know if it's a managed_training or not.

Reply:
hmm by providing default we'd make the usage simpler but it'll be less explicit. either way works